### ISSUE TITLE
sql_cat setting sql_kwargs

### ISSUE THREAD

**Issue by paularmand** [intake/intake-sql#5]
sql_cat makes entries in the catalog much like the yamlfilecatalog does it. The sql-auto receives the sql_kwargs and passes these to dask.dataframe.read_sql and to pandas.read_sql. This breaks for our use case since the cat.describe() gives:
```
{'args': {'index': 'n_hashed',
  'metadata': {'catalog_dir': ''},
  'sql_kwargs': {'name': 'mh_populate_catalog_entries', 'ttl': 1},
  'table': 'Test2',
  'uri': 'mssql+pymssql://user:pass@server/database?charset=utf8'},
 'description': 'SQL table Test2 from mssql+pymssql://user:pass@server/database?charset=utf8',
 'direct_access': True,
 'metadata': {},
 'plugin': 'sql_auto'}
```

pandas.read_sql does not know how to handle these sql_kwargs, so it fails. We can work-around this with:
```
cat.Test2(sql_kwargs={}).read()
```
but this will not pass the kwargs.

If the kwargs should be passed we can test if this works with:
```
cat.Test2(sql_kwargs={'params': {'name': 'mh_populate_catalog_entries', 'ttl': 1}}).read()
```
That works apparently. 

I can PR the code, but unsure why you would want to pass these params to pandas.read_sql. Please advise.

**Comment by martindurant** [comment]
```
'sql_kwargs': {'name': 'mh_populate_catalog_entries', 'ttl': 1}
```
these would appear to be extra kwargs from the general spec of an entry. The following change would fix this
```
--- a/intake_sql/sql_cat.py
+++ b/intake_sql/sql_cat.py
@@ -15,7 +15,8 @@ class SQLCatalog(Catalog):
     name = 'sql_cat'
     version = __version__

-    def __init__(self, uri, views=False, **kwargs):
+    def __init__(self, uri, views=False, sql_kwargs=None, **kwargs):
+        self.sql_kwargs = sql_kwargs or {}
         self.uri = uri
         self.views = views
         super(SQLCatalog, self).__init__(**kwargs)
@@ -32,7 +33,7 @@ class SQLCatalog(Catalog):
                 if c.primary_key:
                     description = 'SQL table %s from %s' % (name, self.uri)
                     args = {'uri': self.uri, 'table': name, 'index': c.name,
-                            'sql_kwargs': self.kwargs}
+                            'sql_kwargs': self.sql_kwargs}
                     e = LocalCatalogEntry(name, description, 'sql_auto', True,
                                           args, {}, {}, {}, "", getenv=False,
                                           getshell=False)
```
with the only problem that anyone who was already passing kwargs and expecting them to get through to read_sql(_table) may be surprised.

The alternative would be to explicitly pluck off kwargs that might describe the source and not be meant for passing on, but this is difficult to future-proof and may fall out of sync with Intake main again.

**Comment by paularmand** [comment]
I believe the main issue for us is the fact that the user will see this (leaky). When we can solve it in the catalog entry that would be okay. If we go with
```
- self.sql_kwargs = sql_kwargs or {}
+ self.sql_kwargs = sql_kwargs or self.kwargs
```
everything would remain backward compatible, and in line with intake, and we can overwrite it ourselves in the catalog entry. okay for PR this way? I'll check the workings during the PR.

**Comment by martindurant** [comment]
If that works, I'd be happy with it, but I'm not certain it will in the case of no override, because those kwargs you were having trouble with will probably appear in `self.kwargs`. Please feel free to try a PR - this will be resolved by the tests!

**Comment by martindurant** [comment]
@paularmand , any further thoughts on this?

**Comment by paularmand** [comment]
I'll try to follow up on this soon.

**Comment by ian-r-rose** [comment]
This is fixed in #6, correct?
